opds2_import.py

```diff
@@ -809,6 +811,58 @@ def _is_open_access_link_(link_data, circulation_data):
 
         return open_access_rights_link
 
+    def _record_coverage_failure(
+        self,
+        failures: Dict[str, List[CoverageFailure]],
+        identifier: Identifier,
+        error_message: str,
+        transient: bool = True,
+    ) -> CoverageFailure:
+        """Record a new coverage failure.
+
+        :param failures: Dictionary mapping publication identifiers to corresponding CoverageFailure objects
+        :type failures: Dict[str, List[CoverageFailure]]
+
+        :param identifier: Publication's identifier
+        :type identifier: Identifier
+
+        :param error_message: Message describing the failure
+        :type error_message: str
+
+        :param transient: Boolean value indicating whether the failure is final or it can go away in the future
+        :type transient: bool
+
+        :return: CoverageFailure object describing the error
+        :rtype: CoverageFailure
+        """
+        if identifier not in failures:
+            failures[identifier.identifier] = []
+
+        failure = CoverageFailure(
+            identifier,
+            error_message,
+            data_source=self.data_source,
+            transient=transient,
+        )
+        failures[identifier.identifier].append(failure)
```

Comment:
Is it possible for a key collision here? The combination of `identifier.type` and `identifier.identifier` is guaranteed to be unique, but `identifier.identifier` alone is not.

This should not be a problem in this PR, since we're allowing only one type of identifier. But it seems possible once the type is generalized.